staging/src/k8s.io/component-base/compatibility/registry.go

```diff
@@ -89,6 +91,8 @@ type ComponentGlobalsRegistry interface {
 	// and cannot be set from cmd flags anymore.
 	// For a given component, its emulation version can only depend on one other component, no multiple dependency is allowed.
 	SetEmulationVersionMapping(fromComponent, toComponent string, f VersionMapping) error
+	// AddMetrics adds metrics for the emulation version of a component.
+	AddMetrics()
```

Comment:
Adding a function to an interface is breaking and this interface is under `staging` so it can be expended on by external projects.

Should be forcing all users of this interface to implement `AddMetrics`?  Looking into how/why we added `SetEmulationVersionMapping` here might add context.

Reply:
Given that this interface was added for compat version and the feature is still in alpha, I'm +1 on adding to the interface since we are providing a very useful feature. I agree with Joe that once we hit beta/GA, these kinds of breaking changes should be considered more carefully, but at the moment I don't think compat version is at that level of stability and is still going through a lot of changes.